### PR TITLE
move exp-asynccache to be a peer dependency

### PR DESCRIPTION
**Pull request by andolf**

### CODE REVIEW

**Review by paed01 — CHANGES_REQUESTED** [review]
I like it. We could also consider dropping the package-lock all-together, lots of bytes.

**Review by markusn — COMMENTED** [review]
Why is this change needed?

**Comment by andolf** [comment]
It's a long time since I did this PR so this is as far as I can remember.
The biggest problem was when I updated `exp-asynccache` and tried to upgrade that package in existing projects: if the project in question used `exp-fetch` then there would exist multiple versions of `exp-asynccache` and at some times it had weird side-effects such as not knowing what API was actually in use. Specifying it as a peer dependency would alleviate some of those problems.

File: .github/workflows/build-latest.yaml [hunk 1]
```diff
@@ -19,5 +19,5 @@ jobs:
         uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node-version }}
-      - run: npm ci
+      - run: npm install
```
Comment (markusn) [inline .github/workflows/build-latest.yaml:22]: I really don't see the point of this. Dropping the package-lock.json and the usage om npm ci means that the "build" will no longer be reproducible

**Review by markusn — CHANGES_REQUESTED** [review]

**Comment by andolf** [comment]
I'm a little bit torn on the implementation:
`exp-fetch` includes a cache by default, which constitutes a requirement to install `exp-asynccache`. However, you can set `cache: null`, but that _still_ requires you to install the cache lib.

**Comment by andolf** [comment]
Closing this because of lack of inactivity

**Pull request closed without merging**